### ISSUE TITLE
Qt/C++ generated code crashes if response is of type string

### ISSUE THREAD

**Issue by paulmasri** [swagger-api/swagger-codegen#4835]
##### Description

Qt/C++ code crashes during the API callback when `output` is cast to `QString*`

On Mac, this causes a fatal SIGSEGV exception in QtCore at `QString::append(QString const&) + 17` according to the crash log.

##### Swagger-codegen version

Current version used by http://editor.swagger.io/#/ at time of posting.

##### Steps to reproduce

Include the following in the yaml:

```
      responses:
        '200':
          description: OK
          schema:
            type: string
            title: jsonData
```

then:

1. Generate Qt/C++ code
2. Create a slot corresponding to the callback and connect the callback signal to it
3. Instantiate the API interface class (e.g. SWGDefaultApi) and call the endpoint

##### Suggest a Fix

I believe the problem is this line of code:

https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/qt5cpp/api-body.mustache#L169

I believe the solution is to detect when `{{returnType}}` is `QString*` and in this case just replace the line with:
```
{{{returnType}}} output = &json;
```

I have tested this locally by replacing (in the generated code)
```
QString* output = static_cast<QString*>(create(json, QString("QString")));
```
with
```
QString* output = &json;
```

**Comment by fehguy** [comment]
@paulmasri yes, this is a defect--returning QString or arrays of QString both have bugs (the latter gives a compile error).

In the mean time, please consider wrapping your QString object in another object and you should have no issue.  But this will get fixed, probably in 2.3.0.

**Comment by wing328** [comment]
@paulmasri if you've time to contribute a fix, please reply to let us know and we'll show you some good starting points.

**Comment by paulmasri** [comment]
@wing328 I'd love to but I just don't have time at the moment. Sorry. In the end I've written my own implementation loosely based on the codegen code. Partly because of this issue and partly because I need HTTP status codes in the response. If I get some slack time (later this year) I'll come back and see if I can be of help.
